src/openforms/formio/rendering/nodes.py

```diff
@@ -42,11 +40,11 @@ class RenderConfiguration:
 @dataclass
 class ComponentNode(Node):
     component: Component
-    step_data: DataMapping  # XXX refactor to FormioData
+    step_data: FormioData
     depth: int = 0
     is_layout = False
-    path: Path | None = None  # Path in the data (#TODO rename to data_path?)
-    json_renderer_path: Path | None = (
+    path: str | None = None  # Path in the data (#TODO rename to data_path?)
```

Comment:
can we further simplify this to 
```suggestion
    path: str = ""  # Path in the data (#TODO rename to data_path?)
```
or is that going to cause issues if someone uses a key like `foo..bar`?

Reply:
I'd prefer that, since always the same type is easier to reason about than even a simple union of types with `None` :)